### PR TITLE
Add options to choose runtime helpers on flattened fields or flattened array elements

### PR DESCRIPTION
**Pull request by a7ehuo**
Add four options as below to choose between runtime
helpers or sym ref generation during the
compilation time for flattened fields or flattened
array elements. E.g. `-Xjit:useFlattenedFieldRuntimeHelpers`
chooses runtime helpers instead of the sym ref generation.

- `useFlattenedArrayElementRuntimeHelpers`
- `useFlattenedFieldRuntimeHelpers`
- `dontUseFlattenedArrayElementRuntimeHelpers`
- `dontUseFlattenedFieldRuntimeHelpers`

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
@Leonardo2718 Ready for review. Thanks!
@hzongaro FYI

File: compiler/control/OMROptions.cpp [hunk 1]
```diff
@@ -1242,6 +1242,8 @@ TR::OptionTable OMR::Options::_jitOptions[] = {
    {"unleashStaticFieldFolding",        "O\tbypass the class white-list, and allow static final fields to be folded aggressively", RESET_OPTION_BIT(TR_RestrictStaticFieldFolding), "F"},
    {"unresolvedSymbolsAreNotColdAtCold", "R\tMark unresolved symbols as cold blocks at cold or lower", SET_OPTION_BIT(TR_UnresolvedAreNotColdAtCold), "F"},
    {"upgradeBootstrapAtWarm",           "R\tRecompile bootstrap AOT methods at warm instead of cold", SET_OPTION_BIT(TR_UpgradeBootstrapAtWarm), "F", NOT_IN_SUBSET},
+   {"useFlattenedFieldArrayRuntimeHelpers",        "M\tuse the runtime helpers for flattened fields for array elements", SET_OPTION_BIT(TR_UseFlattenedFieldArrayRuntimeHelpers), "F"},
+   {"useFlattenedFieldRuntimeHelpers",        "M\tuse the runtime helpers for flattened fields", SET_OPTION_BIT(TR_UseFlattenedFieldRuntimeHelpers), "F"},
```
Comment (Leonardo2718) [inline compiler/control/OMROptions.cpp:1246]: I think there should also be "disable" variants of these options. Maybe `dontUseFlattenedFieldRuntimeHelpers`?

Also I think `useFlattenedArrayElementRuntimeHelpers` is a more accurate name than `useFlattenedFieldArrayRuntimeHelpers`.

Comment (a7ehuo) [reply compiler/control/OMROptions.cpp:1246]: I will add the options to `disable` or not use runtime helper and will update the name for array to `useFlattenedArrayElementRuntimeHelpers` in next commit.

File: compiler/control/OMROptions.cpp [hunk 2]
```diff
@@ -1242,6 +1242,8 @@ TR::OptionTable OMR::Options::_jitOptions[] = {
    {"unleashStaticFieldFolding",        "O\tbypass the class white-list, and allow static final fields to be folded aggressively", RESET_OPTION_BIT(TR_RestrictStaticFieldFolding), "F"},
    {"unresolvedSymbolsAreNotColdAtCold", "R\tMark unresolved symbols as cold blocks at cold or lower", SET_OPTION_BIT(TR_UnresolvedAreNotColdAtCold), "F"},
    {"upgradeBootstrapAtWarm",           "R\tRecompile bootstrap AOT methods at warm instead of cold", SET_OPTION_BIT(TR_UpgradeBootstrapAtWarm), "F", NOT_IN_SUBSET},
+   {"useFlattenedFieldArrayRuntimeHelpers",        "M\tuse the runtime helpers for flattened fields for array elements", SET_OPTION_BIT(TR_UseFlattenedFieldArrayRuntimeHelpers), "F"},
```
Comment (hzongaro) [inline compiler/control/OMROptions.cpp:1245]: I think the thing that holds the value type value is the thing that is flattened - so a "flattened field" or a "flattened array element."  In this case, I think the name should be something like "useFlattenedArrayElementRuntimeHelpers" and for the description "use runtime helpers for flattened array elements"

**Comment by a7ehuo** [comment]
@Leonardo2718  @hzongaro All comments addressed. Ready for another review. Thanks!

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Pull request merged**